test/IRGen/enum.sil

```diff
@@ -2567,15 +2567,14 @@ weak var delegate: delegateProtocol?
 sil @weak_optional : $@convention(thin) (@in StructWithWeakVar?) -> () {
 entry(%x : $*StructWithWeakVar?):
   // CHECK-64:      icmp eq [[WORD]] {{%.*}}, 0
-  // CHECK-64-NEXT: icmp eq [[WORD]] {{%.*}}, 1
   switch_enum_addr %x : $*StructWithWeakVar?,
     case #Optional.some!enumelt.1: a,
     case #Optional.none!enumelt:   b
 a:
   br x
 b:
   // CHECK-64:      store [[WORD]] 0
-  // CHECK-64:      store [[WORD]] 1
+  // CHECK-64:      store [[WORD]] 0
```

Comment:
What caused this change?

Reply:
This change was caused by the consolidation of the extra inhabitance logic between weak and unowned. I thought you were okay with the test fallout given your comment here: https://github.com/apple/swift/pull/16237#pullrequestreview-118134931

That being said, I feel the least confident about this change. Your feedback would be appreciated.